test/py_unit/test_lstm.py

```diff
@@ -0,0 +1,217 @@
+import os
+import sys
+import unittest
+
+import numpy as np
+
+import pytagi.metric as metric
+from examples.data_loader import TimeSeriesDataloader
+from pytagi import Normalizer as normalizer
+from pytagi.nn import LSTM, SLSTM, Linear, OutputUpdater, Sequential, SLinear
+
+# path to binding code
+sys.path.append(
+    os.path.normpath(os.path.join(os.path.dirname(__file__), "..", "..", "build"))
+)
+
+TEST_CPU_ONLY = os.getenv("TEST_CPU_ONLY") == "1"
+
+
+def lstm_test_runner(
+    model: Sequential,
+    input_seq_len: int = 4,
+    batch_size: int = 4,
+    use_cuda: bool = False,
+) -> float:
+    """Run training for time-series forecasting model"""
+    # Dataset
+    output_col = [0]
+    num_features = 1
+    output_seq_len = 1
+    seq_stride = 1
+
+    train_dtl = TimeSeriesDataloader(
+        x_file="data/toy_time_series/x_train_sin_data.csv",
+        date_time_file="data/toy_time_series/train_sin_datetime.csv",
+        output_col=output_col,
+        input_seq_len=input_seq_len,
+        output_seq_len=output_seq_len,
+        num_features=num_features,
+        stride=seq_stride,
+    )
+
+    model.to_device("cuda" if use_cuda else "cpu")
+    out_updater = OutputUpdater(model.device)
+
+    # -------------------------------------------------------------------------#
+    # Training
+    mses = []
+    var_y = np.full((batch_size * len(output_col),), 0.3**2, dtype=np.float32)
+    batch_iter = train_dtl.create_data_loader(batch_size, False)
+
+    for x, y in batch_iter:
```

Comment:
train on 1 epoch, while in `cpp` trains on 2 epochs.

Reply:
I changed from 1 to 2 epochs across all the tests